tests/golos.publication_rewards_tests.cpp

```diff
@@ -880,4 +909,164 @@ BOOST_FIXTURE_TEST_CASE(golos_curation_test, reward_calcs_tester) try {
+    check();
+    interest_rate = 4500;
+    BOOST_CHECK_EQUAL(success(), delegate_vest(_users[5], _users[1], amount, interest_rate));
+    BOOST_CHECK_EQUAL(success(), delegate_vest(_users[6], _users[1], amount, interest_rate));
+    BOOST_CHECK_EQUAL(success(), delegate_vest(_users[7], _users[1], amount, interest_rate));
+    check();
+    BOOST_TEST_MESSAGE("--- " << name{_users[1]}.to_string() << " voted");
+    BOOST_CHECK_EQUAL(success(), addvote(_users[1], {_users[0], "permlink3"}, 10000));
+    check();
+    voter_amount = vest.get_balance_raw(_users[1])["vesting"].as<asset>();
+    auto delegator1_amount = vest.get_balance_raw(_users[5])["vesting"].as<asset>();
+    auto delegator2_amount = vest.get_balance_raw(_users[6])["vesting"].as<asset>();
+    auto delegator3_amount = vest.get_balance_raw(_users[7])["vesting"].as<asset>();
+    produce_blocks(golos::seconds_to_blocks(post.window));
+    BOOST_CHECK_GT(vest.get_balance_raw(_users[1])["vesting"].as<asset>(), voter_amount);
+    BOOST_CHECK_GT(vest.get_balance_raw(_users[5])["vesting"].as<asset>(), delegator_amount);
+    BOOST_CHECK_GT(vest.get_balance_raw(_users[6])["vesting"].as<asset>(), delegator_amount);
+    BOOST_CHECK_GT(vest.get_balance_raw(_users[7])["vesting"].as<asset>(), delegator_amount);
+    check();
+
+    BOOST_TEST_MESSAGE("--- add_funds_to_forum");
+    BOOST_CHECK_EQUAL(success(), add_funds_to_forum(50000));
+    check();
+
+    BOOST_TEST_MESSAGE("--- several delegators interest_rate > 0 with one interest_rate == 0");
+    BOOST_TEST_MESSAGE("--- create_message: " << name{_users[0]}.to_string());
+    BOOST_CHECK_EQUAL(success(), create_message({_users[0], "permlink4"}));
+    check();
+    interest_rate = 4500;
+    
+    BOOST_CHECK_EQUAL(success(), delegate_vest(_users[8], _users[1], amount, interest_rate));
+    BOOST_CHECK_EQUAL(success(), delegate_vest(_users[9], _users[1], amount, interest_rate));
+    BOOST_CHECK_EQUAL(success(), delegate_vest(_users[10], _users[1], amount, interest_rate0));
+    check();
+    BOOST_TEST_MESSAGE("--- " << name{_users[1]}.to_string() << " voted");
+    BOOST_CHECK_EQUAL(success(), addvote(_users[1], {_users[0], "permlink4"}, 10000));
+    check();
+    voter_amount = vest.get_balance_raw(_users[1])["vesting"].as<asset>();
+    delegator1_amount = vest.get_balance_raw(_users[8])["vesting"].as<asset>();
+    delegator2_amount = vest.get_balance_raw(_users[9])["vesting"].as<asset>();
+    delegator3_amount = vest.get_balance_raw(_users[10])["vesting"].as<asset>();
+    produce_blocks(golos::seconds_to_blocks(post.window));
+    BOOST_CHECK_GT(vest.get_balance_raw(_users[1])["vesting"].as<asset>(), voter_amount);
+    BOOST_CHECK_GT(vest.get_balance_raw(_users[8])["vesting"].as<asset>(), delegator_amount);
+    BOOST_CHECK_GT(vest.get_balance_raw(_users[9])["vesting"].as<asset>(), delegator_amount);
+    BOOST_CHECK_EQUAL(vest.get_balance_raw(_users[10])["vesting"].as<asset>(), delegator_amount);
+    check();
```

Comment:
Propose to add 1 more test:
+ curator have no vesting
+ delegator have lot of vesting and 100% rate

In this case `vote.delegator.interest_rate` becomes 100% (please check in test) and curator gets no reward. It's useful to be sure upper edge works as expected (do not overflows beyond 100%)